pkg/await/apps_deployment.go

```diff
@@ -403,6 +441,7 @@ func (dia *deploymentInitAwaiter) processDeploymentEvent(event watch.Event) {
 	}
 
 	dia.checkReplicaSetStatus()
+	dia.checkPersistentVolumeClaimStatus()
```

Comment:
If we start bubbling up errors from other types, calling these `*Status` methods in the right places will start to get confusing. @lblackstone any chance we can simplify this a bit in your upcoming refactoring?

Reply:
Yes, this is one of the main things I'd like to tackle next. Won't be in the client-go refactor, but that will enable a more state-machine based approach.

Reply:
Per @lblackstone's comment, it seems like this is a branch of work in and of itself. I'll table this feedback for the sake of this PR

Reply:
Ah, to be clear, I'd meant mainly to bring this to Levi's attention rather than to ask you to do it. :)